agg-ast/schema_derivation/src/match_schema_derivation.rs

```diff
@@ -1345,26 +1410,22 @@ impl MatchConstrainSchema for Expression {
             Ok(())
         }
 
+        /// match_derive_replace constarins the result set schema for $replace operators in match. The
```

Comment:
Sadly constarins is too powerful, even for schema schema derivation!

```suggestion
        /// match_derive_replace constrains the result set schema for $replace operators in match. The
```